hw_diag/diagnostics/lte_diagnostic.py

```diff
@@ -54,45 +56,43 @@ def perform_test(self, diagnostics_report):
             LOGGER.error(f"Error while retrieving list of LTE devices: {e}")
             diagnostics_report.record_failure(e, self)
 
-    def process_lte_devices(self, lte_devices, diagnostics_report):
+    def process_lte_devices(self, lte_devices: list, diagnostics_report: DiagnosticsReport) -> None:
         if len(lte_devices) > 0:
             diagnostics_report.record_result(lte_devices, self)
         else:
-            diagnostics_report.record_failure(NO_LTE_DEVICES_MSG, self)
-
-
-def get_lte_devices():
-    lte_devices = []
-    system_bus = dbus.SystemBus()
-    proxy = system_bus.get_object(DBUS_MM1_SERVICE, DBUS_MM1_PATH)
-    manager = dbus.Interface(proxy, DBUS_OBJECTMANAGER)
-    modems = manager.GetManagedObjects()
-
-    for modem in modems:
-        append_lte_device_from_modem(lte_devices, modem, system_bus)
-
-    LOGGER.info(f"Found the following LTE Devices: {lte_devices}")
-    return lte_devices
-
-
-def append_lte_device_from_modem(lte_devices, modem, system_bus):
-    modem_proxy = system_bus.get_object(DBUS_MM1_SERVICE, modem)
-    props_iface = dbus.Interface(modem_proxy, DBUS_PROPERTIES)
-    props = props_iface.GetAll(DBUS_MM1_IF_MODEM)
-
-    model = props.get("Model")
-    manufacturer = props.get("Manufacturer")
-    capabilities = props.get("CurrentCapabilities")
-    equipment_id = props.get("EquipmentIdentifier")
-
-    is_lte = capabilities & MM_MODEM_CAPABILITY['LTE'] or \
-        capabilities & MM_MODEM_CAPABILITY['LTE_ADVANCED']
-
-    if is_lte:
-        lte_devices.append({
-            "Model": str(model),
-            "Manufacturer": str(manufacturer),
-            "EquipmentIdentifier": str(equipment_id)
-        })
-
-    return lte_devices
+            diagnostics_report.record_failure(self.NO_LTE_DEVICES_MSG, self)
+
+    def get_lte_devices(self) -> list:
+        lte_devices = []
+        system_bus = dbus.SystemBus()
+        proxy = system_bus.get_object(self.DBUS_MM1_SERVICE, self.DBUS_MM1_PATH)
+        manager = dbus.Interface(proxy, self.DBUS_OBJECTMANAGER)
+        modems = manager.GetManagedObjects()
+
+        for modem in modems:
+            self.append_lte_device_from_modem(lte_devices, modem, system_bus)
+
+        LOGGER.info(f"Found the following LTE Devices: {lte_devices}")
+        return lte_devices
+
+    def append_lte_device_from_modem(self, lte_devices: list, modem, system_bus) -> list:
```

Comment:
Missing some type hints

Reply:
Now, all the methods of the all diagnostics are type hinted.